ExtrusionPath::polyline conversion from Polyline to Polyline3 breaks Debug configuration

### Is this issue reproducible in the latest nightly build?

- [x] I have checked the latest nightly build and the issue is still reproducible

### Is there an existing issue for this problem?

- [x] I have searched the existing issues

### OrcaSlicer Version

2.4.0-dev

### Operating System (OS)

Linux, macOS

### OS Version

Tahoe 26.5

### Additional system information

_No response_

### Printer

Snapmaker U1

### How to reproduce

Try to compile a debug mode executable:



### Actual results

```
In file included from /Users/me/proj/OrcaSlicer/src/slic3r/Utils/RaycastManager.cpp:1:
In file included from /Users/me/proj/OrcaSlicer/build/arm64/src/slic3r/CMakeFiles/libslic3r_gui.dir/Debug/cmake_pch.hxx:4:
In file included from /Users/me/proj/OrcaSlicer/src/slic3r/pchheader.hpp:34:
In file included from /Users/me/proj/OrcaSlicer/deps/build/OrcaSlicer_dep/usr/local/include/eigen3/Eigen/Dense:1:
In file included from /Users/me/proj/OrcaSlicer/deps/build/OrcaSlicer_dep/usr/local/include/eigen3/Eigen/Core:352:
/Users/me/proj/OrcaSlicer/deps/build/OrcaSlicer_dep/usr/local/include/eigen3/Eigen/src/Core/CwiseBinaryOp.h:93:3: error: static assertion failed due to requirement '(int(Eigen::internal::size_of_xpr_at_compile_time<Eigen::Matrix<long long, 2, 1, 2, 2, 1>>::ret) == 0 && int(Eigen::internal::size_of_xpr_at_compile_time<Eigen::Matrix<long long, 3, 1, 2, 3, 1>>::ret) == 0) || ((int(Matrix<long long, 2, 1, 2, 2, 1>::RowsAtCompileTime) == Eigen::Dynamic || int(Matrix<long long, 3, 1, 2, 3, 1>::RowsAtCompileTime) == Eigen::Dynamic || int(Matrix<long long, 2, 1, 2, 2, 1>::RowsAtCompileTime) == int(Matrix<long long, 3, 1, 2, 3, 1>::RowsAtCompileTime)) && (int(Matrix<long long, 2, 1, 2, 2, 1>::ColsAtCompileTime) == Eigen::Dynamic || int(Matrix<long long, 3, 1, 2, 3, 1>::ColsAtCompileTime) == Eigen::Dynamic || int(Matrix<long long, 2, 1, 2, 2, 1>::ColsAtCompileTime) == int(Matrix<long long, 3, 1, 2, 3, 1>::ColsAtCompileTime)))': YOU_MIXED_MATRICES_OF_DIFFERENT_SIZES
   93 |   EIGEN_STATIC_ASSERT_SAME_MATRIX_SIZE(Lhs, Rhs)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```


This is due to the type mismatch in ExtrusionLoop::verify that's wrapped in NDEBUG


```
#ifndef NDEBUG
	bool validate() const {
		assert(this->first_point() == this->paths.back().polyline.points.back());
		for (size_t i = 1; i < paths.size(); ++ i)
			assert(this->paths[i - 1].polyline.points.back() == this->paths[i].polyline.points.front());
		return true;
	}
#endif /* NDEBUG */

```



### Expected results

Compiles as expected

### Project file & Debug log uploads

No project file is associated with this code bug

### Checklist of files to include

- [ ] Log file
- [ ] Project file

### Anything else?

The recent change to introduce Z Anti Aliasing (ZAA) changed ExtrusionPath::polyline from Polyline to Polyline3. This validation code was missed as part of that change

--- a/src/libslic3r/ExtrusionEntity.hpp
+++ b/src/libslic3r/ExtrusionEntity.hpp
@@ -118,6 +118,9 @@ public:
     virtual void reverse() = 0;
     virtual Point first_point() const = 0;
     virtual Point last_point() const = 0;
+    virtual const Point3& first_point3() const = 0;
+    virtual const Point3& last_point3() const = 0;
+    
     // Produce a list of 2D polygons covered by the extruded paths, offsetted by the extrusion width.
     // Increase the offset by scaled_epsilon to achieve an overlap, so a union will produce no gaps.
     virtual void polygons_covered_by_width(Polygons &out, const float scaled_epsilon) const = 0;
@@ -253,9 +256,9 @@ public:
 	ExtrusionEntity* clone_move() override { return new ExtrusionPath(std::move(*this)); }
     void reverse() override { this->polyline.reverse(); }
     Point first_point() const override { return this->polyline.points.front().to_point(); }
-    Point3 first_point3() const { return this->polyline.points.front(); }
+    const Point3& first_point3() const override { return this->polyline.points.front(); }
     Point last_point() const override { return this->polyline.points.back().to_point(); }
-    Point3 last_point3() const { return this->polyline.points.back(); }
+    const Point3& last_point3() const override { return this->polyline.points.back(); }
     size_t size() const { return this->polyline.size(); }
     bool empty() const { return this->polyline.empty(); }
     bool is_closed() const { return ! this->empty() && this->polyline.points.front() == this->polyline.points.back(); }
@@ -403,7 +406,9 @@ public:
 	ExtrusionEntity* clone_move() override { return new ExtrusionMultiPath(std::move(*this)); }
     void reverse() override;
     Point first_point() const override { return this->paths.front().polyline.points.front().to_point(); }
+    const Point3& first_point3() const override { return this->paths.front().polyline.points.front(); }
     Point last_point() const override { return this->paths.back().polyline.points.back().to_point(); }
+    const Point3& last_point3() const override { return this->paths.back().polyline.points.back(); }
     size_t size() const { return this->paths.size(); }
     bool empty() const { return this->paths.empty(); }
     double length() const override;
@@ -459,7 +464,9 @@ public:
     bool is_counter_clockwise() { return this->polygon().is_counter_clockwise(); }
     void reverse() override;
     Point first_point() const override { return this->paths.front().polyline.points.front().to_point(); }
+    const Point3& first_point3() const override { return this->paths.front().polyline.points.front(); }
     Point last_point() const override { assert(this->first_point() == this->paths.back().polyline.points.back().to_point()); return this->first_point(); }
+    const Point3& last_point3() const override { assert(this->first_point3() == this->paths.back().polyline.points.back()); return this->first_point3(); }
     Polygon polygon() const;
     double length() const override;
     bool split_at_vertex(const Point &point, const double scaled_epsilon = scaled<double>(0.001));
@@ -510,7 +517,7 @@ public:
 
 #ifndef NDEBUG
 	bool validate() const {
-		assert(this->first_point() == this->paths.back().polyline.points.back());
+		assert(this->first_point3() == this->paths.back().polyline.points.back());
 		for (size_t i = 1; i < paths.size(); ++ i)
 			assert(this->paths[i - 1].polyline.points.back() == this->paths[i].polyline.points.front());
 		return true;

--- a/src/libslic3r/ExtrusionEntityCollection.hpp
+++ b/src/libslic3r/ExtrusionEntityCollection.hpp
@@ -119,7 +119,10 @@ public:
     	{ return this->no_sort ? *this : chained_path_from(this->entities, start_near, role); }
     void reverse() override;
     Point first_point() const override { return this->entities.front()->first_point(); }
+    const Point3& first_point3() const override { return this->entities.front()->first_point3(); }
     Point last_point() const override { return this->entities.back()->last_point(); }
+    const Point3& last_point3() const override { return this->entities.back()->last_point3(); }
+
     // Produce a list of 2D polygons covered by the extruded paths, offsetted by the extrusion width.
     // Increase the offset by scaled_epsilon to achieve an overlap, so a union will produce no gaps.
     void polygons_covered_by_width(Polygons &out, const float scaled_epsilon) const override;

